zebra-consensus/src/block/check.rs

```diff
@@ -90,5 +90,5 @@ pub fn subsidy_is_correct(network: Network, block: &Block) -> Result<(), BoxErro
             return Ok(());
         }
     }
-    Err("subsidy is not correct")?
+    Err("validation of block subsidy rules inside coinbase transaction failed")?
```

Comment:
We try to format errors as "error kind: conditions that caused the error".

The checks in the function are incomplete, so it is hard to write a good error description. Here is an error with all the checks that the function should do. As you can see, the error is very long.

So we should split up the checks for:
* the sum of the coinbase outputs, and
* the founders reward.

Since we want to check the founders reward for blocks 1..Canopy, but always check the coinbase output sum, these checks should be in different functions.

```suggestion
    Err("invalid coinbase transaction: the sum of the coinbase transaction outputs must be less than or equal to the block subsidy plus transaction fees, and the exact founders reward value must be sent as a single output to the correct address")?
```

Reply:
We can't do the `sum of coinbase outputs` check yet, so let's just check the founders reward, and error if it fails.

Then we can add a TODO and `Ok(())` at the end of the function